include/ignition/gui/GuiEvents.hh

```diff
@@ -43,6 +43,158 @@ namespace ignition
         /// \brief Unique type for this event.
         static const QEvent::Type kType = QEvent::Type(QEvent::MaxUser);
       };
+
+      /// \brief The class for sending and receiving custom snap value events.
+      class SnapIntervals : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _xyz XYZ snapping values.
+        /// \param[in] _rpy RPY snapping values.
+        /// \param[in] _scale Scale snapping values.
+        public: SnapIntervals(
+                    const math::Vector3d &_xyz,
+                    const math::Vector3d &_rpy,
+                    const math::Vector3d &_scale)
+        : QEvent(kType), xyz(_xyz), rpy(_rpy), scale(_scale)
+        {
+        }
+  
+        /// \brief Get the XYZ snapping values.
+        /// \return The XYZ snapping values.
+        public: math::Vector3d XYZ() const
+        {
+          return this->xyz;
+        }
+  
+        /// \brief Get the RPY snapping values.
+        /// \return The RPY snapping values.
+        public: math::Vector3d RPY() const
+        {
+          return this->rpy;
+        }
+  
+        /// \brief Get the scale snapping values.
+        /// \return The scale snapping values.
+        public: math::Vector3d Scale() const
+        {
+          return this->scale;
+        }
+  
+        /// \brief The QEvent representing a snap event occurrence.
+        static const QEvent::Type kType = QEvent::Type(QEvent::User);
+  
+        /// \brief XYZ snapping values in meters, these values must be positive.
+        private: math::Vector3d xyz;
+  
+        /// \brief RPY snapping values in degrees, these values must be positive.
+        private: math::Vector3d rpy;
+  
+        /// \brief Scale snapping values - a multiplier of the current size,
+        /// these values must be positive.
+        private: math::Vector3d scale;
+      };
+
+      /// \brief Event called to spawn a preview model.
+      /// Used by plugins that spawn models.
+      class SpawnPreviewModel : public QEvent
+      {
+        /// \brief Constructor
+        /// \param[in] _modelSdfString The model's SDF file as a string.
```

Comment:
Same here about Ignition GUI not knowing about SDF, let's say something like "the model's description as a string"

Reply:
be5a482d8da3da972dff28a187723c805378d367